src/javascript/app/pages/user/metatrader/metatrader.ui.js

```diff
@@ -598,15 +605,33 @@ const MetaTraderUI = (() => {
             trading_server.id === accounts_info[account].info.server
         );
 
+    const shouldSetTradingPassword = () => {
+        const { status } = State.getResponse('get_account_status');
+        if (Array.isArray(status)) {
+            status.includes('trading_password_required');
+        } else {
+            throw new Error('get_account_status is undefined');
+        }
+    };
+
     const displayStep = (step) => {
         const new_account_type = newAccountGetType();
         const is_demo = /demo/.test(new_account_type);
+        const should_set_trading_password = shouldSetTradingPassword();
         const is_synthetic = /gaming/.test(new_account_type);
 
         $form.find('#msg_form').remove();
         $form.find('#mv_new_account div[id^="view_"]').setVisibility(0);
         $form.find(`#view_${step}`).setVisibility(1);
         $form.find('#view_3').find('.error-msg, .days-to-crack').setVisibility(0);
+
+        // Show proper notice msg based on api flag
+        if (should_set_trading_password) {
+            $form.find('#view_3').find('#trading_password_new_user').setVisibility(1);
+        } else {
+            $form.find('#view_3').find('#trading_password_existing_user').setVisibility(1);
```

Comment:
I created a new user, on my first login when I tried to create a `Real`->`Synthetic` account on MT5, I saw ` If you've forgotten your password, click Reset password` sentence which user must see it when the password is set before not on first time he is trying to set.